proton

```diff
@@ -1267,16 +1284,40 @@ class Session:
         if "MEDIACONV_VIDEO_TRANSCODED_FILE" in self.env:
             f.write("\tMEDIACONV_VIDEO_TRANSCODED_FILE=\"" + self.env["MEDIACONV_VIDEO_TRANSCODED_FILE"] + "\" \\\n")
 
-    def dump_dbg_scripts(self):
+    def get_steam_runtime_launch_client_command(self, launcher_address):
+        launch_client = os.path.expanduser("~/.steam/root/ubuntu12_32/steam-runtime/amd64/bin/steam-runtime-launch-client")
+
+        # This relies on an implementation detail of
+        # SteamLinuxRuntime_soldier, but is necessary until the scout
+        # runtime shipped with the GA Steam client has been updated to
+        # include a copy of s-r-launch-client.
+        if "PRESSURE_VESSEL_RUNTIME_BASE" in self.env and not os.path.exists(launch_client):
+            launch_client = self.env["PRESSURE_VESSEL_RUNTIME_BASE"] + "/pressure-vessel/bin/steam-runtime-launch-client"
+
+        return "".join([
+            launch_client, " \\\n",
+            "\t", shell_quote(launcher_address), " \\\n",
+            # --directory='' means keep using the game's working directory
+            "\t--directory='' \\\n",
+            "\t-- \\\n",
+        ])
+
+    def dump_dbg_scripts(self, *, launcher_address=""):
         exe_name = os.path.basename(sys.argv[2])
 
         tmpdir = self.env.get("PROTON_DEBUG_DIR", "/tmp") + "/proton_" + os.environ["USER"] + "/"
         makedirs(tmpdir)
 
+        if launcher_address:
+            launch_client = self.get_steam_runtime_launch_client_command(launcher_address) + "\tenv \\\n"
+        else:
+            launch_client = ""
+
         with open(tmpdir + "winedbg", "w") as f:
             f.write("#!/bin/bash\n")
             f.write("#Run winedbg with args\n\n")
             f.write("cd \"" + os.getcwd() + "\"\n")
+            # TODO: Should this run via launch_client?
```

Comment:
Since I'm still unsure about this, I've moved this part into a second commit, so that the bit I'm more confident about (the first commit) can be considered separately.

In the second commit, I've changed it to use IPC for `winedbg` and `winedbg_run` but not `run`. Other suggestions welcome...

Reply:
I'm no longer altering these scripts.